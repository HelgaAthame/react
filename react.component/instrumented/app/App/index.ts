function cov_9oz6m35w1() {
  const path = 'C:\\react\\first\\helgaathame-JSFE2022Q3\\react.component\\src\\app\\App\\index.ts';
  const hash = '5d12842a76c368ab1753d867a5f437eee1b81513';
  const global = new Function('return this')();
  const gcv = '__coverage__';
  const coverageData = {
    path: 'C:\\react\\first\\helgaathame-JSFE2022Q3\\react.component\\src\\app\\App\\index.ts',
    statementMap: {},
    fnMap: {},
    branchMap: {},
    s: {},
    f: {},
    b: {},
    _coverageSchema: '1a1c01bbd47fc00a2c39e90264f33305004495a9',
    hash: '5d12842a76c368ab1753d867a5f437eee1b81513',
  };
  const coverage = global[gcv] || (global[gcv] = {});
  if (!coverage[path] || coverage[path].hash !== hash) {
    coverage[path] = coverageData;
  }
  const actualCoverage = coverage[path];
  {
    // @ts-ignore
    cov_9oz6m35w1 = function () {
      return actualCoverage;
    };
  }
  return actualCoverage;
}
cov_9oz6m35w1();
export * from './App';
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJuYW1lcyI6WyJjb3ZfOW96Nm0zNXcxIiwiYWN0dWFsQ292ZXJhZ2UiXSwic291cmNlcyI6WyJpbmRleC50cyJdLCJzb3VyY2VzQ29udGVudCI6WyJleHBvcnQgKiBmcm9tICcuL0FwcCc7XHJcbiJdLCJtYXBwaW5ncyI6Ijs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7OztJQWVZO0lBQUFBLGFBQUEsWUFBQUEsQ0FBQTtNQUFBLE9BQUFDLGNBQUE7SUFBQTtFQUFBO0VBQUEsT0FBQUEsY0FBQTtBQUFBO0FBQUFELGFBQUE7QUFmWixjQUFjLE9BQU8ifQ==
