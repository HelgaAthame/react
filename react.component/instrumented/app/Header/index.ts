function cov_2kfjon2d8p() {
  const path =
    'C:\\react\\first\\helgaathame-JSFE2022Q3\\react.component\\src\\app\\Header\\index.ts';
  const hash = 'd2473c65f591f5b98051910ab3551cd772d9fba5';
  const global = new Function('return this')();
  const gcv = '__coverage__';
  const coverageData = {
    path: 'C:\\react\\first\\helgaathame-JSFE2022Q3\\react.component\\src\\app\\Header\\index.ts',
    statementMap: {},
    fnMap: {},
    branchMap: {},
    s: {},
    f: {},
    b: {},
    _coverageSchema: '1a1c01bbd47fc00a2c39e90264f33305004495a9',
    hash: 'd2473c65f591f5b98051910ab3551cd772d9fba5',
  };
  const coverage = global[gcv] || (global[gcv] = {});
  if (!coverage[path] || coverage[path].hash !== hash) {
    coverage[path] = coverageData;
  }
  const actualCoverage = coverage[path];
  {
    // @ts-ignore
    cov_2kfjon2d8p = function () {
      return actualCoverage;
    };
  }
  return actualCoverage;
}
cov_2kfjon2d8p();
export * from './Header';
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJuYW1lcyI6WyJjb3ZfMmtmam9uMmQ4cCIsImFjdHVhbENvdmVyYWdlIl0sInNvdXJjZXMiOlsiaW5kZXgudHMiXSwic291cmNlc0NvbnRlbnQiOlsiZXhwb3J0ICogZnJvbSAnLi9IZWFkZXInO1xyXG4iXSwibWFwcGluZ3MiOiI7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7SUFlWTtJQUFBQSxjQUFBLFlBQUFBLENBQUE7TUFBQSxPQUFBQyxjQUFBO0lBQUE7RUFBQTtFQUFBLE9BQUFBLGNBQUE7QUFBQTtBQUFBRCxjQUFBO0FBZlosY0FBYyxVQUFVIn0=
