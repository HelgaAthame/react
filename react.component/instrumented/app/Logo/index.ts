function cov_1qdg42jfls() {
  var path = "C:\\react\\first\\helgaathame-JSFE2022Q3\\react.component\\src\\app\\Logo\\index.ts";
  var hash = "3263ab11fb758aff4ce735d3539b60ca03930297";
  var global = new Function("return this")();
  var gcv = "__coverage__";
  var coverageData = {
    path: "C:\\react\\first\\helgaathame-JSFE2022Q3\\react.component\\src\\app\\Logo\\index.ts",
    statementMap: {},
    fnMap: {},
    branchMap: {},
    s: {},
    f: {},
    b: {},
    _coverageSchema: "1a1c01bbd47fc00a2c39e90264f33305004495a9",
    hash: "3263ab11fb758aff4ce735d3539b60ca03930297"
  };
  var coverage = global[gcv] || (global[gcv] = {});
  if (!coverage[path] || coverage[path].hash !== hash) {
    coverage[path] = coverageData;
  }
  var actualCoverage = coverage[path];
  {
    // @ts-ignore
    cov_1qdg42jfls = function () {
      return actualCoverage;
    };
  }
  return actualCoverage;
}
cov_1qdg42jfls();
export * from './Logo';
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJuYW1lcyI6WyJjb3ZfMXFkZzQyamZscyIsImFjdHVhbENvdmVyYWdlIl0sInNvdXJjZXMiOlsiaW5kZXgudHMiXSwic291cmNlc0NvbnRlbnQiOlsiZXhwb3J0ICogZnJvbSAnLi9Mb2dvJztcclxuIl0sIm1hcHBpbmdzIjoiOzs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7O0lBZVk7SUFBQUEsY0FBQSxZQUFBQSxDQUFBO01BQUEsT0FBQUMsY0FBQTtJQUFBO0VBQUE7RUFBQSxPQUFBQSxjQUFBO0FBQUE7QUFBQUQsY0FBQTtBQWZaLGNBQWMsUUFBUSJ9