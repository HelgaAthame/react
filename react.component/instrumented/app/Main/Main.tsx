import { useEffect } from 'react';
import './main.scss';
import { BookType } from '../types/';
import { Card } from '../Card';
import { Loading } from '../Loading';

import { useDispatch, useSelector } from 'react-redux';
import { AppDispatch, RootState } from '../../redux-folder';
import { fetchChars } from '../../redux-folder';

export const Main = () => {
  const { cards, error, loading, searchText } = useSelector((state: RootState) => state.curState);

  const dispatch = useDispatch<AppDispatch>();

  useEffect(() => {
    dispatch(fetchChars(searchText)); // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [dispatch]);

  return (
    <div className="main" data-testid="main">
      {cards && !loading && cards.map((doc: BookType) => <Card key={doc._id} {...doc} />)}
      {loading && <Loading />}
      {error && <div className="error">Error: {error}</div>}
    </div>
  );
};
