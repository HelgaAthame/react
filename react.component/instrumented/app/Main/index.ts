function cov_1196sh5ubb() {
  var path = "C:\\react\\first\\helgaathame-JSFE2022Q3\\react.component\\src\\app\\Main\\index.ts";
  var hash = "8135d1be7892b84123b3e300140d1f4ad3c6243f";
  var global = new Function("return this")();
  var gcv = "__coverage__";
  var coverageData = {
    path: "C:\\react\\first\\helgaathame-JSFE2022Q3\\react.component\\src\\app\\Main\\index.ts",
    statementMap: {},
    fnMap: {},
    branchMap: {},
    s: {},
    f: {},
    b: {},
    _coverageSchema: "1a1c01bbd47fc00a2c39e90264f33305004495a9",
    hash: "8135d1be7892b84123b3e300140d1f4ad3c6243f"
  };
  var coverage = global[gcv] || (global[gcv] = {});
  if (!coverage[path] || coverage[path].hash !== hash) {
    coverage[path] = coverageData;
  }
  var actualCoverage = coverage[path];
  {
    // @ts-ignore
    cov_1196sh5ubb = function () {
      return actualCoverage;
    };
  }
  return actualCoverage;
}
cov_1196sh5ubb();
export * from './Main';
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJuYW1lcyI6WyJjb3ZfMTE5NnNoNXViYiIsImFjdHVhbENvdmVyYWdlIl0sInNvdXJjZXMiOlsiaW5kZXgudHMiXSwic291cmNlc0NvbnRlbnQiOlsiZXhwb3J0ICogZnJvbSAnLi9NYWluJztcclxuIl0sIm1hcHBpbmdzIjoiOzs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7O0lBZVk7SUFBQUEsY0FBQSxZQUFBQSxDQUFBO01BQUEsT0FBQUMsY0FBQTtJQUFBO0VBQUE7RUFBQSxPQUFBQSxjQUFBO0FBQUE7QUFBQUQsY0FBQTtBQWZaLGNBQWMsUUFBUSJ9