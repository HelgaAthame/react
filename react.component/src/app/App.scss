#root {
  max-width: 1280px;
  margin: 0 auto;
  padding: 2rem;
  text-align: center;

}
.app {
  display: flex;
  flex-direction: column;
  .header {

  }
  .main {
    width: 100vw;
    display: flex;
    flex-direction: row;
    flex-wrap:wrap;
  }
}
