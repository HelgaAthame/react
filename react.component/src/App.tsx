import './App.scss'
import SearchBar from './SearchBar.js'

function App() {
  //const [count, setCount] = useState(0)

  history.replaceState(null, '', "main")
  return (
    <div className="App">
      <SearchBar />
    </div>
  )
}

export default App
