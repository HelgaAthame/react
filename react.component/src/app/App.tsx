import { Component } from 'react';
import './App.scss';
import Header from './Header.js';
import Main from './Main.js';
import cards, { CardT } from './cards';
import SearchBar from './SearchBar';

class App extends Component {
  state = {
    cards: cards,
  };

  updateData(cards: CardT[]) {
    this.setState({ cards: cards });
  }

  render() {
    return (
      <div className="app">
        <Header cards={[]}>
          <SearchBar cards={this.state.cards} updateData={this.updateData.bind(this)} />
        </Header>
        <Main cards={this.state.cards} updateData={this.updateData.bind(this)}/>
      </div>
    );
  }
}

export default App;
