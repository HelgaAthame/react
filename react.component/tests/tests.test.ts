/*
 * @vitest-environment
happy-dom
 */

import { mount } from "@vue/test-utils";
import {
  beforeEach,
  describe,
  expect,
  test,
} from "vitest";
import { render, screen, fireEvent } from '@testing-library/react';

import App from '../src/app/App';
import cards from '../src/app/cards'
import Header from "../src/app/Header";
import Logo from "../src/app/Logo";
import Main from "../src/app/Main";

const props = {};
const app = new App(props);

beforeEach( () => {
  //app.setState({cards: cards});
})

describe('react app', () => {

  test('cards imports to app instance correctly', () => {
    //const props = {};
    //const app = new App(props);
    expect(app.state.cards).toEqual(cards);
  });

  test('order of books is proper', () => {
    //const props = {};
    //const app = new App(props);
    expect(app.state.cards[2]).toMatchInlineSnapshot(`
      {
        "author": "Franz Kafka",
        "country": "Czech",
        "genre": "novel",
        "likes": 3,
        "name": "The Castle",
        "picture": "https://upload.wikimedia.org/wikipedia/commons/thumb/e/ed/Franz_Kafka_Das_Schloss.jpg/220px-Franz_Kafka_Das_Schloss.jpg",
        "price": 1200,
        "year": 1926,
      }
    `);
  });

  test('all the card field import properly', () => {
    //const props = {};
    //const app = new App(props);
    expect(app.state.cards[1]).toMatchSnapshot(`
    Object {
      "name": "The Modern JavaScript Tutorial",
      "country": "Russia",
      "author: "Ilya Kantor",
      "genre": "tutorial",
      "likes": 2,
      "picture": "https://i.livelib.ru/workpic/1001593289/200/a57d/Ilya_Kantor__Sovremennyj_uchebnik_JavaScript.jpg"
    }
    `);
  });

  test('should update App State', () => {
    const fakeCards = [
      {
        name: 'Holly Bible',
        price: 999,
        country: 'Fake Country 1',
        genre: 'fake genre',
        year: 1000,
        author: 'Jesus Christ',
        likes: -5,
        picture: 'fake URL 1',
      },
      {
        name: 'Fake Name 2',
        price: 555,
        country: 'Fake Country 2',
        genre: 'fake genre',
        year: 0,
        author: 'Santa Claus',
        likes: Math.PI,
        picture: 'fake URL 2'
      }
    ];

    app.state = {cards: fakeCards};
    expect(app.state.cards[0].author).toBe('Jesus Christ');
    expect(app.state.cards[0].name).toBe('Holly Bible');
  });

  test('logo renders', () => {
    render(<Logo />);
    //WHY DOESN'T LOGO RENDERS???????//
  })
})
