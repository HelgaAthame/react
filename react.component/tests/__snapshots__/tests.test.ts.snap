// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`react app > all the card field import properly > 
    Object {
      "name": "The Modern JavaScript Tutorial",
      "country": "Russia",
      "author: "Ilya Kantor",
      "genre": "tutorial",
      "likes": 2,
      "picture": "https://i.livelib.ru/workpic/1001593289/200/a57d/Ilya_Kantor__Sovremennyj_uchebnik_JavaScript.jpg"
    }
     1`] = `
{
  "author": "Ilya Kantor",
  "country": "Russia",
  "genre": "tutorial",
  "likes": 2,
  "name": "The Modern JavaScript Tutorial",
  "picture": "https://i.livelib.ru/workpic/1001593289/200/a57d/Ilya_Kantor__Sovremennyj_uchebnik_JavaScript.jpg",
  "price": 59,
  "year": 2007,
}
`;
